cmd/bucket-notification-handlers.go

```diff
@@ -43,6 +43,10 @@ const (
 // not enabled on the bucket, the operation returns an empty
 // NotificationConfiguration element.
 func (api objectAPIHandlers) GetBucketNotificationHandler(w http.ResponseWriter, r *http.Request) {
+	if api.gateway {
+		writeErrorResponse(w, ErrNotImplemented, r.URL)
+		return
+	}
```

Comment:
you can let the underlying object layer return the error if it does not implement this call.

Reply:
as discussed, for the time being this check is being left as is. Can be removed after notifications are moved to object layer and a SetNotificationConfig method is in place.